pkg/cmd/kafka/list/list.go

```diff
@@ -185,6 +180,43 @@ func mapResponseItemsToRows(kafkas []kafkamgmtclient.KafkaRequest, selectedId st
 	return rows
 }
 
+func getClusterIdMapFromKafkas(opts *options, kafkas []kafkamgmtclient.KafkaRequest) (map[string]*v1.Cluster, error) {
+	kafkaClusterIds := make([]string, len(kafkas))
+	for _, kafka := range kafkas {
+		if kafka.GetClusterId() != "" {
+			kafkaClusterIds = append(kafkaClusterIds, kafka.GetClusterId())
+		}
+	}
```

Comment:
You could further improve this by making sure that the IDs are unique.. but it is good enough already

Reply:
I tried to think about this, how would you make sure the IDs are unique in a way that would be faster then what happens now which is the chance to add the same id and insert the same key, value pair into the map?

I tried to think of a solution but I am not sure.

Reply:
> value pair into the map?

a `Set` data type is what would have been suitable here. But golang doesn't have that and a map can be used instead.  IM0, what you've now is okay as at the API is capable of handling that via the search query that's constructed.

Reply:
Done